src/App.tsx

```diff
@@ -1,24 +1,293 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import {
+  FC,
+  ChangeEvent,
+  FormEvent,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import './App.scss';
 import { UserWarning } from './UserWarning';
+import {
+  getTodos,
+  removeTodo,
+  addTodoToServer,
+  changeTodo,
+} from './api/todos';
+import { TodosList } from './components/TodosList/TodosList';
+import { Todo } from './types/Todo';
+import { ErrorInfo } from './components/ErrorInfo/ErrorInfo';
+import { visibleTodos, getcompletedTodosIds } from './utils/todoUtils';
+import { StatusValue } from './types/StatusValue';
 
-const USER_ID = 0;
+const USER_ID = 10725;
+
+export const App: FC = () => {
+  const formRef = useRef<HTMLInputElement | null>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [queryTodo, setQueryTodo] = useState('');
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([]);
+  const [statusTodo, setstatusTodo] = useState<StatusValue>(StatusValue.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [visibleError, setVisibleError] = useState('');
+
+  const completedTodosIds = getcompletedTodosIds(todos);
+  const isAllTodosCompleted = todos.every(todo => todo.completed);
+
+  const getTodosFromServer = async () => {
+    try {
+      const fetchedTodos = await getTodos(USER_ID);
+
+      setTodos(fetchedTodos as Todo[]);
+    } catch (error) {
+      setVisibleError('Unable to load a todos');
+    }
+  };
+
+  useEffect(() => {
+    if (formRef.current) {
+      formRef.current.focus();
+    }
+
+    getTodosFromServer();
+  }, [tempTodo, loadingTodos, visibleError]);
 
-export const App: React.FC = () => {
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const removesTodo = async (todosId: number[]) => {
+    try {
+      setLoadingTodos(prevIds => [...prevIds, ...todosId]);
+
+      await Promise.all(
+        todosId.map(async id => {
+          await removeTodo(id);
+        }),
+      );
+
+      const updatedTodos = todos.filter(todo => !todosId.includes(todo.id));
+
+      setTodos(updatedTodos);
+    } catch (error) {
+      setVisibleError('Unable to delete a todo');
+    } finally {
+      setLoadingTodos([]);
+    }
+  };
+
+  const addTodo = async (title: string) => {
+    try {
+      const newTodo = {
+        title: title.trim(),
+        completed: false,
+        userId: USER_ID,
+      };
+
+      const tempId = 0;
+
+      setTempTodo({
+        ...newTodo,
+        id: tempId,
+      });
+      setIsInputDisabled(true);
+      setLoadingTodos([tempId]);
```

Comment:
What if we removed a todo before adding a new one?

Reply:
@silchyn  I don't understand, it works fine if I do that...

can you explain what you mean? and how I can check it?

Reply:
@silchyn got it, maybe))) on the next review check it)